### ISSUE TITLE
Tabs are not appropriately marked up

### ISSUE THREAD

**Issue by nicklemmon** [SparkPost/matchbox#217]
This is a fairly complex update that would be required, and unfortunately, because the tabs appear to be uncontrolled at this point, some of the changes might be tough to implement. I'm all ears on how that might be possible without rocking the boat too much.

## Implementation

* See the official [WAI-ARIA working example](https://www.w3.org/TR/wai-aria-practices-1.1/examples/tabs/tabs-1/tabs.html)
* See the [Reach UI Tabs](https://ui.reach.tech/tabs/) for a React implementation

Tabs are structured like so - there are lots of `aria` attributes to leverage to make this work, and some fancy keyboard interaction required as well:

```html
<div>
  <div role="tablist">
    <button
      id="tab-1"
      role="tab"
      aria-selected="true"
      tabindex="0"
      aria-controls="panel-1"
    >
      Tab 1 - I am currently selected
    </button>

    <button
      id="tab-2"
      role="tab"
      aria-selected="false"
      tabindex="-1"
      aria-controls="panel-2"
    >
      Tab 2
    </button>

    <button
      id="tab-3"
      role="tab"
      aria-selected="false"
      tabindex="-1"
      aria-controls="panel-3"
    >
      Tab 3
    </button>
  </div>

  <div>
    <div
      role="tabpanel"
      aria-labelledby="tab-1"
      style="display: block;"
    >
      <p>I am content controlled by tab 1</p>
    </div>

    <div
      role="tabpanel"
      aria-labelledby="tab-2"
      style="display: none;"
    >
      <p>I am content controlled by tab 2</p>
    </div>

    <div
      role="tabpanel"
      aria-labelledby="tab-3"
      style="display: none;"
    >
      <p>I am content controlled by tab 3</p>
    </div>
  </div>
</div>
```

Note that because of the associations using `id` attributes via `aria-controls`, the tab panels cannot be unmounted - they must be hidden instead. This can be accomplished in a few ways, all of these are roughly equivalent from an a11y POV:

* Add and remove CSS classes that toggle visibility
* Use inline styles
* Use the `hidden` attribute

When using inline styles or a separate stylesheet, be sure that hidden tabs are set to `display: none;` to make sure screen readers do not encounter content inappropriately.

**Comment by jonambas** [comment]
This is a breaking change for sure. Will require changes to 2web2ui as well.

Some AC for when we do the work:
- Tabs should be accessible
- Tab state should be controlled internally and provide hooks for – default state, on select, etc

**Comment by nicklemmon** [comment]
Hey, @jonambas - this one is pretty stale at this point - any chance there is capacity for it?

**Comment by jonambas** [comment]
@nicklemmon 

Hmm, I was thinking that the best way to implement this would be through a reusable hook. We already have `useTabs`, we could modify it to provide `tabpanel` attributes as well. 

Another option is to create a new Tabs component from the ground up – we would need to change how some of the tabs work in the UI. IIRC we have some tabs that function as links.

**Comment by nicklemmon** [comment]
@jonambas Not sure why I only just now saw your comment here. Sorry about that!

I would think two things would be needed:
- Update the component API to make it composable - Reach UI has [a good example](https://reach.tech/tabs/)
- Add a custom hook - a "headless" implementation would probably be best

In general, if a tab is a link then they shouldn't semantically be tabs (though visually they could still look like tabs). I'm mainly thinking the tags on the `/join` page in which clicking on them really does take the user to an entirely new page/place.

Other tabs that happen to update the route when clicked (like on the `/domains` page) could still be tabs semantically while controlling the route.
